pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -525,13 +547,23 @@ func (r *ReconcileClusterSync) Reconcile(ctx context.Context, request reconcile.
 		}
 	}
 
-	result := reconcile.Result{Requeue: true, RequeueAfter: r.timeUntilFullReapply(lease)}
-	if syncSetsNeedRequeue || selectorSyncSetsNeedRequeue {
+	result := reconcile.Result{Requeue: true, RequeueAfter: r.timeUntilRenew(lease)}
+	if syncSetsNeedRequeue == requeueWithDelay || selectorSyncSetsNeedRequeue == requeueWithDelay {
+		result.RequeueAfter = time.Second * time.Duration(10)
+	} else if syncSetsNeedRequeue == requeueImmediately || selectorSyncSetsNeedRequeue == requeueImmediately {
```

Comment:
I wonder if these conditions should be reversed -- i.e. if either needs to be requeued immediately, we should do that rather than delaying.

(That said, I'm still not convinced delaying is The Right Thing™.)